### PR TITLE
Improve error handling and propagate exceptions back to driver

### LINKED ISSUE #1726: [Bug]: The onError calls are always ignored
### What happened?

Any send operation which expects response can specify criteria for matching answers but also error handlers which should be called when handling logic fails. Determination when error handler should be called is not trivial. Most of protocols have fairly long `unwrap`/`check` statements which can fail at any stage. We can not assume which stage will fail, we can't layer error handlers, thus we need carefully craft logic and make sure that we don't catch too much and still propagate failures to driver and clean up handlers.
Currently it might be that packet consumers are retained until timeout, even if we know that they can't properly handle incoming data.

### Version

v0.12

### Programming Languages

- [X] plc4j
- [ ] plc4go
- [ ] plc4c
- [ ] plc4net

### Protocols

- [ ] AB-Ethernet
- [ ] ADS /AMS
- [ ] BACnet/IP
- [ ] CANopen
- [ ] DeltaV
- [ ] DF1
- [ ] EtherNet/IP
- [ ] Firmata
- [ ] KNXnet/IP
- [ ] Modbus
- [ ] OPC-UA
- [ ] S7

### PR DESCRIPTION
**Pull request by splatch**
Closes #1726.

### CODE REVIEW

File: plc4j/spi/src/main/java/org/apache/plc4x/java/spi/ConversationContext.java [hunk 1]
```diff
@@ -149,7 +149,7 @@ interface ContextHandler {
 
         void cancel();
 
-        void awaitResponse() throws InterruptedException, ExecutionException;
+        void await() throws InterruptedException, ExecutionException;
```
Comment (chrisdutz) [inline plc4j/spi/src/main/java/org/apache/plc4x/java/spi/ConversationContext.java:152]: Why change this? I think the awaitResponse gave context to what you're waiting for?

**Review by chrisdutz — APPROVED** [review]
LGTM (Looks Good To Me)

File: plc4j/spi/src/main/java/org/apache/plc4x/java/spi/ConversationContext.java (hunk 1 above)
Comment (splatch) [reply plc4j/spi/src/main/java/org/apache/plc4x/java/spi/ConversationContext.java:152]: We use same context for server side/simulators. This statement is applicable to `expectRequest()` thus renaming it  to `await` makes it more generic and match various roles implied by use of API.

Comment (chrisdutz) [reply plc4j/spi/src/main/java/org/apache/plc4x/java/spi/ConversationContext.java:152]: Ups... Didn't want to unresolve.. Mobile phone UI slighty clunky. No objections

**Pull request merged**